packages/components/src/internal/components/forms/actions.ts

```diff
@@ -399,3 +402,38 @@ export function getUsersWithPermissions(permissions: string | string[] = Permiss
         });
     });
 }
+
+export type UsersLoader = (permissions: string | string[]) => Promise<List<User>>;
+
+interface UsersState {
+    error: string;
+    loadingState: LoadingState;
+    users: User[];
+}
+
+export function useUsersWithPermissions(
```

Comment:
Very cool. This is a good opportunity to write a unit test for a hook. I gave this a go with `useTabLinks` in `@labkey/workflow` and it was a little challenging. I ended up declaring a wrapping test component that invoked the hook and then testing the results of the hook by querying/interacting with that component. Curious to see what approach you take.

Reply:
Added some tests. Did the same thing, created a component that renders the state, and also checked that the loader was getting called with the proper args.